cmake/kokkos_tribits.cmake

```diff
@@ -241,6 +241,37 @@ function(KOKKOS_SET_LIBRARY_PROPERTIES LIBRARY_NAME)
     target_link_options(${LIBRARY_NAME} PUBLIC ${KOKKOS_LINK_OPTIONS})
   endif()
 
+  if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.18)
+    if(KOKKOS_LINK_OPTIONS)
+      include(CheckLinkerFlag)
+      #exclude case of compiler_launcher forwarding to nvcc_wrapper as the used CXX compiler is shadowed in this case (compiler_launcher changes the compiler).
+      #The CXX compiler CMake will invoke for the check is not able to consume the cuda flags.
+      if(NOT (KOKKOS_ENABLE_CUDA) OR ("${CMAKE_CXX_COMPILER}" MATCHES "nvcc_wrapper") OR (${KOKKOS_CXX_COMPILER_ID}
```

Comment:
Can you add some parentheses, I find the expression quite hard to read.
Can the second half be expressed as `NOT` matching `compiler_launcher` to reflect the comment above?

Reply:
unfortunately not. The compile launcher gets set after this function, so we don't know yet here.